src/api/web/index.ts

```diff
@@ -78,6 +78,24 @@ export default function(config, auth, storage) {
     });
   }
 
+  // Favicon
+  const customFaviconPath = _.get(config, 'web.favicon') || '';
+  router.get('/-/mapped/favicon', function(req, res): void {
+    if (customFaviconPath) {
+      return res.redirect('/-/mapped/custom-favicon');
+    } else {
+      return res.redirect('/-/static/favicon.ico');
+    }
+  });
+  router.get('/-/mapped/custom-favicon', function(req, res): void {
+    if (fs.existsSync(customFaviconPath)) {
```

Comment:
I'd rather use https://nodejs.org/api/fs.html#fs_fs_access_path_mode_callback 

Here an example, https://github.com/verdaccio/monorepo/blob/9.x/plugins/local-storage/src/local-fs.ts#L209

Access asynchronously is much more performant, we don't block the main thread. Otherwise, anyone could abuse of this endpoint and block the whole server, let's put that weight on the event loop instead.